src/components/action-popover/action-popover.component.js

```diff
@@ -0,0 +1,251 @@
+import React, {
+  useState, useCallback, useEffect, useRef
+} from 'react';
+import PropTypes from 'prop-types';
+import I18n from 'i18n-js';
+import {
+  Menu, MenuItemFactory, MenuButton, ButtonIcon, MenuItemIcon, MenuItemDivider
+} from './action-popover.style';
+import Events from '../../utils/helpers/events';
+import createGuid from '../../utils/helpers/guid';
+import OptionsHelper from '../../utils/helpers/options-helper';
+
+const ActionPopover = ({ children, id }) => {
+  const [isOpen, setOpen] = useState(false);
+  const [focusIndex, setFocusIndex] = useState(0);
+  const [childrenWithRef, setChildrenWithRef] = useState();
+  const [items, setItems] = useState();
+  const [guid] = useState(createGuid());
+  const button = useRef();
+
+  // Keyboard commands implemented as reccomended by WAI-ARIA best practices
+  // https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-actions.html
+
+  const onButtonKeyDown = useCallback(((e) => {
+    if (Events.isSpaceKey(e) || Events.isDownKey(e) || Events.isEnterKey(e)) {
+      e.preventDefault();
+      setOpen(!isOpen);
+      if (isOpen) {
+        button.current.focus();
+      } else {
+        setFocusIndex(0);
+      }
+    } else if (Events.isUpKey(e)) {
+      e.preventDefault();
+      setOpen(true);
```

Comment:
Looking at the examples in the w3 link commented above, shouldn't we be setting the focusIndex here to the last item in the menu?

https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-actions.html#kbd1_label

>Up Arrow | Opens menu and moves focus to last menuitem

Reply:
Good spot! I was using https://www.w3.org/TR/2016/WD-wai-aria-practices-1.1-20161214/examples/menu-button/menu-button-1/menu-button-1.html previously where the wording is slightly different.

`Open menu and move focus to previous menuitem`
I interpreted this wrong, which is probably why they have changed the wording.

I'll make a fix.

Reply:
Fixed in edaf0af030189a540afb12a5209a1ab61e4146bc